### ISSUE TITLE
silhouette not calculated correctly for pam

### ISSUE THREAD

**Issue by ze97286** [Nik-Zainal-Group/signature.tools.lib#46]
[line](https://github.com/Nik-Zainal-Group/signature.tools.lib/blob/master/R/SignatureExtractionLib.R#L626) seems like a copy paste bug: 
 
```
 sil_hclust <- summary(cluster::silhouette(cut_res,as.dist(distMatrix)))
 sil_pam <- summary(cluster::silhouette(cut_res,as.dist(distMatrix)))
```
should it not be using cut_res.PAM for pam clustering?

**Comment by andreadega** [comment]
Thanks for spotting this, I have added a fix to the upcoming 2.2.0 version, should be out in the next few days.

**Comment by andreadega** [comment]
This should be solved in the 2.2.0 version now public.
